modules/graceful-restart.adoc

```diff
@@ -0,0 +1,117 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/graceful-cluster-restart.adoc
+
+[id="graceful-restart_{context}"]
+= Restarting the cluster
+
+You can restart your cluster after it has been shut down gracefully.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* This procedure assumes that you gracefully shut down the cluster.
+
+.Procedure
+
+. Power on any cluster dependencies, such as external storage or an LDAP server.
+
+. Start all cluster machines.
++
+Use the appropriate method for your cloud environment to start the machines, for example, from your cloud provider's web console.
++
+Wait approximately 10 minutes before continuing to check the status of master nodes.
+
+. Verify that all master nodes are ready.
++
+----
+$ oc get nodes -l node-role.kubernetes.io/master
+----
++
+The master nodes are ready if the status is `Ready`, as shown in the following output:
++
+----
+NAME                           STATUS   ROLES    AGE   VERSION
+ip-10-0-168-251.ec2.internal   Ready    master   75m   v1.18.3
+ip-10-0-170-223.ec2.internal   Ready    master   75m   v1.18.3
+ip-10-0-211-16.ec2.internal    Ready    master   75m   v1.18.3
+----
+
+. After the master nodes are ready, verify that all worker nodes are ready.
++
+----
+$ oc get nodes -l node-role.kubernetes.io/worker
+----
++
+The worker nodes are ready if the status is `Ready`, as shown in the following output:
++
+----
+NAME                           STATUS   ROLES    AGE   VERSION
+ip-10-0-179-95.ec2.internal    Ready    worker   64m   v1.18.3
+ip-10-0-182-134.ec2.internal   Ready    worker   64m   v1.18.3
+ip-10-0-250-100.ec2.internal   Ready    worker   64m   v1.18.3
+----
+
+. If the worker nodes are _not_ ready, then check whether there are any pending certificate signing requests that must be approved.
```

Comment:
> If the worker nodes
This sentence seems not only applies worker nodes, applies to master nodes as well.

Reply:
Good point. @vareti @sttts Is there a chance that the master nodes will need to have CSRs approved before they become ready? Like, that we need to add the steps to check for and approve CSRs for the masters to step 3?

https://osdocs-1086-shutdown--ocpdocs.netlify.app/openshift-enterprise/latest/backup_and_restore/graceful-cluster-restart.html#graceful-restart_graceful-restart-cluster

Reply:
@xingxingxia is right. I just came to know that it is possible. We could have similar steps to check CSR for master nodes too.